### PR TITLE
#972: Implement token registration 

### PR DESCRIPTION
**Pull request by andreoss**
Per #972 
- Add methods 
- Implement User's token registration

### CODE REVIEW

**Comment by charlesmike** [comment]
@andreoss thank you for your Pull Request. I'll assign someone to review it soon.

**Comment by charlesmike** [comment]
@amihaiemil please review this Pull Request. Deadline (when it should be merged or closed) is ``2021-02-13T02:23:44.783748``.

You should check if the requirements have been implemented (partially or in full), if there are unit tests covering the changes and if the CI build passes. Feel free to reject the PR or ask for changes if it's too big or not clear enough.

Estimation here is ``30 minutes``, that's how much you will be paid. You will be paid even if this PR gets rejected.

File: self-core-impl/src/main/java/com/selfxdsd/core/BaseSelf.java [hunk 1]
```diff
@@ -167,6 +167,16 @@ public ApiTokens apiTokens() {
             return this.user.apiTokens();
         }
 
+        @Override
+        public ApiToken register(
```
Comment (amihaiemil) [inline self-core-impl/src/main/java/com/selfxdsd/core/BaseSelf.java:171]: @andreoss This method is not needed here. We will register the new token through the User, like this:

```java
final User authenticated = this.sell.login(Login);
final ApiToken registered = authenticated.register(name, token, expiration);
```

File: self-core-impl/src/main/java/com/selfxdsd/core/managers/StoredProjectManager.java [hunk 2]
```diff
@@ -644,5 +644,16 @@ public ApiTokens apiTokens() {
...
         }
+
+        @Override
+        public ApiToken register(
+            final String name,
+            final String token,
+            final LocalDateTime expiration
+        ) {
+            throw new UnsupportedOperationException(
+                "A Project Manager can not register ApiTokens!"
```
Comment (amihaiemil) [inline self-core-impl/src/main/java/com/selfxdsd/core/managers/StoredProjectManager.java:655]: "A Project manager cannot *have* ApiTokens".

**Review by amihaiemil — CHANGES_REQUESTED** [review]
@andreoss looks good. Just 2 comments

**Comment by amihaiemil** [comment]
@andreoss I removed the unneeded method :D

**Comment by amihaiemil** [comment]
@andreoss I put that method back. Sorry my mistake. I thought it was a method of BaseSelf, but it's in fact in BaseSelf.Authenticated. 

So it's ok.

**Comment by charlesmike** [comment]
@amihaiemil thank you for resolving this ticket. I've just added it to your active invoice. You can always check all your invoices and more on the [Contributor Dashboard](https://self-xdsd.com).

**Pull request merged**